### PR TITLE
Feature/customer verification

### PR DESCRIPTION
**Pull request by sandile-shongwe**
-  Added Material Checkbox to the CDK
- Created the screen for customer verification
- Wrote preliminary unit tests

### CODE REVIEW

File: projects/sbgcc-sales/src/lib/sbgcc-sales-routing.module.ts [hunk 1]
```diff
@@ -4,13 +4,13 @@ import { RouterModule, Routes } from '@angular/router';
 import { CustomerOfferComponent } from './components/customer-offer/customer-offer.component';
 import { CustomerOrderScreenComponent } from './components/customer-order-screen/customer-order-screen.component';
 import { CustomerSearchComponent } from './components/customer-search/customer-search.component';
-
-
+import { CustomerVerificationComponent } from './components/customer-verification/customer-verification.component';
 
 const routes: Routes = [
   { path: 'sales/customer-search', component: CustomerSearchComponent },
```
Comment (andrevanzyl1) [inline projects/sbgcc-sales/src/lib/sbgcc-sales-routing.module.ts]: In all the routing modules after login, pls add "canActivate: [AuthGuard]"
e.g.
{ path: 'sales/customer-search', component: CustomerSearchComponent, canActivate: [AuthGuard] },

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<div class="flex-container verification-header">
```
Comment (andrevanzyl1) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.html]: You can remove all empty spaces and the use "format document" option

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  correctCount: number;
+  verified = false;
+  currentCustomer: CustomerModel;
+  isEnabled = false;
+  constructor(protected authService: AuthenticationService,
+    protected cacheService: CacheService,
+    protected messageService: MessageService,
+    protected navigationService: NavigationService) {
+    super(cacheService, messageService);
+    this.currentCustomer = {
```
Comment (andrevanzyl1) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: This data will need to be replaced from the customer search screen

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      id: '9602254077099',
+      maritalStatus: 'Single',
+      residentialAddress: '1 complex,  Drive, Suburb, City',
+      alternativeNumber: '0114765099'
+    };
+  }
+
+  initialize() {
+    this.correctCount = 0;
+  }
```
Comment (andrevanzyl1) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: To get the customer from navigation, use the below example code
initialize(): void {
this.correctCount = 0;
    const navigationParameters = this.getNavigationParameter<CustomerModel>(true);
    if (navigationParameters) {
this.currentCustomer = navigationParameters;
}
}

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts (hunk 3 above)
Comment (sandile-shongwe) [reply projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: done

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+import { AuthenticationService } from 'sbgcc-common-services';
+import { BaseComponent, CacheService, MessageService, NavigationService } from 'sbgcc-core';
+
+import { CustomerModel } from '../../models/customer.model';
+
+@Component({
+  selector: 'app-customer-verification',
+  templateUrl: './customer-verification.component.html',
+  styleUrls: ['./customer-verification.component.scss']
```
Comment (TianeErwee) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: Please add encapsulation: ViewEncapsulation.None

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+})
+export class CustomerVerificationComponent extends BaseComponent {
+
+  correctCount: number;
+  verified = false;
+  currentCustomer: CustomerModel;
+  isEnabled = false;
```
Comment (TianeErwee) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: You do not need this boolean, it always has the same value as verified so you can use verified wherever you use isEnabled

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  initialize() {
+    this.correctCount = 0;
+  }
+
+  nextButtonClicked() {}
+
+  verifiedStatus(checked: boolean) {
+    if (checked) {
```
Comment (TianeErwee) [inline projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: We can simplify this like so:

this.correctCount = this.correctCount + 1*(checked ? 1 : -1);
this.verified = !!(this.correctCount >=6);

File: projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts (hunk 5 above)
Comment (sandile-shongwe) [reply projects/sbgcc-sales/src/lib/components/customer-verification/customer-verification.component.ts]: Somehow when I use ViewEncapsulation None on the component the checkboxes shrink to like almost a line.
Please assist. I'll also look into it.

**Pull request merged**